core/testing.go

```diff
@@ -67,40 +42,13 @@ func StartTestNode(t *testing.T) testnode.Context {
 }
 
 // StartTestNodeWithConfig starts Tendermint and Celestia App tandem with custom configuration.
-func StartTestNodeWithConfig(t *testing.T, cfg *TestConfig) testnode.Context {
-	state, kr, err := testnode.DefaultGenesisState(cfg.Accounts...)
-	require.NoError(t, err)
-
-	tmNode, app, cctx, err := testnode.New(
-		t,
-		cfg.ConsensusParams,
-		cfg.Tendermint,
-		cfg.SuppressLogs,
-		state,
-		kr,
-		"private",
-	)
-	require.NoError(t, err)
-
-	cctx, cleanupCoreNode, err := testnode.StartNode(tmNode, cctx)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		err := cleanupCoreNode()
-		require.NoError(t, err)
-	})
-
-	cctx, cleanupGRPCServer, err := StartGRPCServer(app, cfg.App, cctx)
-	require.NoError(t, err)
-	t.Cleanup(func() {
-		err := cleanupGRPCServer()
-		require.NoError(t, err)
-	})
-
+func StartTestNodeWithConfig(t *testing.T, cfg *testnode.Config) testnode.Context {
+	cctx, _, _ := testnode.NewNetwork(t, cfg)
```

Comment:
Do you cleanup the started network with t underneath?

Reply:
yeah

Reply:
we can remove more code, as there's a bit of duplicated code here that is handled by the app
https://github.com/celestiaorg/celestia-app/blob/965eafb4357376aec31f84f3628f7703c5587f9a/test/util/testnode/full_node.go#L149-L198